BrainPortal/app/models/cluster_task.rb

```diff
@@ -2350,6 +2330,47 @@ def validate_json_string json_string
     JSON::Validator.validate!(schema,json_string) # raises an exception if json_string is not valid
   end
 
+  ##################################################################
+  # Singularity load
+  ##################################################################
+
+  private
+
+  def container_image_name
+    ".container-#{self.run_id}.img"
+  end
+
+  # Install the singularity image as a local file.
+  def load_singularity_image #:nodoc:
+    singularity_image_name = self.tool_config.containerhub_image_name
+
+    # Singularity PULL
+    if singularity_image_name.present?
+      begin
+        errfile = ".container_load_cmd.#{self.run_id}.err"
+        success = system("#{singularity_executable_name} pull --name #{container_image_name} #{singularity_image_name.bash_escape}  </dev/null >/dev/null 2>#{errfile.bash_escape}")
+        err     = File.read(errfile) rescue ""
+        cb_error "Cannot pull singularity image" if err.present? || !success
+        return
+      ensure
+        File.unlink(errfile)    rescue true
+      end
+    end
+
+    # Singularity registration (local image)
+    if singularity_image = self.tool_config.container_image # = not ==
+      # Create a link to our image; the image is a registered CBRAIN file
+      self.addlog("Syncing the singularity image")
+      singularity_image    = self.tool_config.container_image
```

Comment:
This assignment to the variable `singularity_image` repeats exactly what is in the IF statement of the block